app/controllers/confirmations_controller.rb

```diff
@@ -11,18 +11,34 @@ def show
   end
 
   def confirm
+    confirmation_token = params[resource_name][:confirmation_token]
+    @resource = resource
     digested_token = Devise.token_generator
-                         .digest(self, :confirmation_token, params[resource_name][:confirmation_token])
+                         .digest(self, :confirmation_token, confirmation_token)
 
     self.resource = resource_class.find_by_confirmation_token(digested_token) if
-        params[resource_name][:confirmation_token].present?
+        confirmation_token.present?
+
+    if update_attributes && match_passwords
+      self.resource = resource_class.confirm_by_token(confirmation_token)
 
-    if resource.update_attributes(params[resource_name].except(:confirmation_token).permit(:email, :password, :password_confirmation)) && resource.password_match?
-      self.resource = resource_class.confirm_by_token(params[resource_name][:confirmation_token])
       set_flash_message :notice, :confirmed
       sign_in_and_redirect(resource_name, resource)
     else
-      render action: 'show'
+      render action: :show
     end
   end
+
+  private
+
+  def match_passwords
+    @resource.password_match?
+  end
+
+  def update_attributes
+    attributes = params[resource_name].except(:confirmation_token)
+                     .permit(:email, :password, :password_confirmation)
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Align the operands of an expression in an assignment spanning multiple lines.